modules/database-commons/src/main/java/org/testcontainers/ext/ScriptUtils.java

```diff
@@ -285,21 +287,43 @@ public static boolean containsSqlScriptDelimiters(String script, String delim) {
 		return false;
 	}
 
-	/**
-	 * Load script from classpath and apply it to the given database
-	 *
-	 * @param databaseDelegate database delegate for script execution
-	 * @param initScriptPath   the resource to load the init script from
-	 */
-	public static void runInitScript(DatabaseDelegate databaseDelegate, String initScriptPath) {
-		try {
-			URL resource = ScriptUtils.class.getClassLoader().getResource(initScriptPath);
+    /**
+     * Load script from classpath and apply it to the given database
+     *
+     * @param databaseDelegate database delegate for script execution
+     * @param initScriptPath   the resource to load the init script from
+     */
+    public static void runInitScript(DatabaseDelegate databaseDelegate, String initScriptPath) {
+        runInitScript(databaseDelegate, initScriptPath, DEFAULT_STATEMENT_SEPARATOR);
+    }
+
+    /**
+     * Load script from classpath and apply it to the given database
+     *
+     * @param databaseDelegate    database delegate for script execution
+     * @param initScriptPath      the resource to load the init script from
+     * @param initScriptSeparator separator to split script
+     */
+    public static void runInitScript(DatabaseDelegate databaseDelegate, String initScriptPath,
+                                     String initScriptSeparator) {
+        try {
+            URL resource;
+            if (initScriptPath.startsWith(FILE_PATH_PREFIX)) {
```

Comment:
this part was moved from `ContainerDatabaseDriver` to make this method single-contract and reusable in three places